rom/dev/src/pcr.rs

```diff
@@ -39,26 +55,96 @@ pub fn extend_pcr0(env: &RomEnv) -> CaliptraResult<()> {
     // Lock the PCR from clear
     pcr_bank.map(|p| p.set_pcr_lock(caliptra_drivers::PcrId::PcrId0));
 
-    let extend = |data: Array4x12| {
+    let extend = |data: Array4x12, pcr_entry_id: PcrLogEntryId| {
         let bytes: &[u8; 48] = &data.into();
-        sha.map(|s| pcr_bank.map(|p| p.extend_pcr(PcrId::PcrId0, s, bytes)))
+        sha.map(|s| pcr_bank.map(|p| p.extend_pcr(PcrId::PcrId0, s, bytes)))?;
+        log_pcr(pcr_entry_id, PcrId::PcrId0, bytes, 0)
     };
 
-    let extend_u8 = |data: u8| {
+    let extend_u8 = |data: u8, pcr_entry_id: PcrLogEntryId| {
         let bytes = &data.to_le_bytes();
-        sha.map(|s| pcr_bank.map(|p| p.extend_pcr(PcrId::PcrId0, s, bytes)))
+        sha.map(|s| pcr_bank.map(|p| p.extend_pcr(PcrId::PcrId0, s, bytes)))?;
+        log_pcr(pcr_entry_id, PcrId::PcrId0, bytes, 0)
     };
 
-    extend_u8(env.dev_state().map(|d| d.lifecycle()) as u8)?;
-    extend_u8(env.dev_state().map(|d| d.debug_locked()) as u8)?;
-    extend_u8(env.fuse_bank().map(|f| f.anti_rollback_disable()) as u8)?;
-    extend(env.fuse_bank().map(|f| f.vendor_pub_key_hash()))?;
-    extend(env.data_vault().map(|d| d.owner_pk_hash()))?;
-    extend_u8(env.data_vault().map(|d| d.vendor_pk_index()) as u8)?;
-    extend(env.data_vault().map(|d| d.fmc_tci()))?;
-    extend_u8(env.data_vault().map(|d| d.fmc_svn()) as u8)?;
+    extend_u8(
+        env.dev_state().map(|d| d.lifecycle()) as u8,
+        PcrLogEntryId::DeviceLifecycle,
+    )?;
+    extend_u8(
+        env.dev_state().map(|d| d.debug_locked()) as u8,
+        PcrLogEntryId::DebugLocked,
+    )?;
+    extend_u8(
+        env.fuse_bank().map(|f| f.anti_rollback_disable()) as u8,
+        PcrLogEntryId::AntiRollbackDisabled,
+    )?;
+    extend(
+        env.fuse_bank().map(|f| f.vendor_pub_key_hash()),
+        PcrLogEntryId::VendorPubKeyHash,
+    )?;
+    extend(
+        env.data_vault().map(|d| d.owner_pk_hash()),
+        PcrLogEntryId::OwnerPubKeyHash,
+    )?;
+    extend_u8(
+        env.data_vault().map(|d| d.vendor_pk_index()) as u8,
+        PcrLogEntryId::VendorPubKeyIndex,
+    )?;
+    extend(env.data_vault().map(|d| d.fmc_tci()), PcrLogEntryId::FmcTci)?;
+    extend_u8(
+        env.data_vault().map(|d| d.fmc_svn()) as u8,
+        PcrLogEntryId::FmcSvn,
+    )?;
 
     // TODO: Check PCR0 != 0
 
     Ok(())
 }
+
+/// Log PCR data
+///
+/// # Arguments
+/// * `pcr_entry_id` - PCR log entry ID
+/// * `pcr_id` - PCR ID
+/// * `data` - PCR data
+/// * `metadata` - PCR metadata
+///
+/// # Return Value
+/// * `Ok(())` - Success
+/// * `Err(GlobalErr::PcrLogInvalidEntryId)` - Invalid PCR log entry ID
+/// * `Err(GlobalErr::PcrLogUpsupportedDataLength)` - Unsupported data length
+///
+pub fn log_pcr(
+    pcr_entry_id: PcrLogEntryId,
+    pcr_id: PcrId,
+    data: &[u8],
+    _metadata: u32,
```

Comment:
Optional: consider removing this for now. I suspect that when it gets wired up to `STASH_MEASUREMENT`, it will not be needed. Can always add it back without messing with the PCR log format.

Reply:
Remmoved.